core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,290 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ComponentParser {
+    private static ComponentParser INSTANCE;
+
+    private static Map<String, SecurityScheme> securitySchemeMap;
+    private static Long versionNumber;
+    private static String connectionName;
+
+
+    private ComponentParser() {}
+
+
+    public synchronized static ComponentParser getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ComponentParser();
+        }
+        return INSTANCE;
+    }
+
+    public List<Component> parse(OpenAPI openAPI) {
+        List<Component> components = new ArrayList<>();
+        securitySchemeMap = openAPI.getComponents().getSecuritySchemes();
+        versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        connectionName = openAPI.getInfo().getTitle();
+
+
+
+        Paths paths = openAPI.getPaths();
+
+        for (String pathName : paths.keySet()) {
+            PathItem path = paths.get(pathName);
+            Map<PathItem.HttpMethod, Operation> operations = path.readOperationsMap();
+
+            for (PathItem.HttpMethod httpMethod : operations.keySet()) {
+                components.addAll(initComponents(operations.get(httpMethod), httpMethod, pathName));
+            }
+
+        }
+
+       return components;
+    }
+
+    private  List<Component> initComponents(Operation operation, PathItem.HttpMethod operationName, String pathName) {
+        List<Component> components = new ArrayList<>();
+        if (operation != null) {
+
+            Set<String> securities = getSecurities(operation.getParameters(), operation.getSecurity());
+            Set<String> requestContents = getRequestContents(operation.getRequestBody());
+            Set<String> responseContents = getResponseContents(operation.getResponses());
+
+            List<HashMap<String, String>>  names = generateName(securities, requestContents, responseContents);
+
+            if (!names.isEmpty()) {
+                for (HashMap<String, String> partNames : names) {
+                    components.add(createComponent(partNames, operation, operationName, pathName));
+                }
+            } else {
+                components.add(createComponent(new HashMap(), operation, operationName, pathName));
+            }
+        }
+        return components;
+    }
+
+
+    private  Set<String> getSecurities(List<io.swagger.v3.oas.models.parameters.Parameter> parameters, List<SecurityRequirement> securitiesList) {
+        Set<String> securities = new HashSet<>();
+        if (parameters != null) {
+            securities = parameters.stream().map(parameter -> {
+                if (parameter.getIn() != null && securitySchemeMap.containsKey(parameter.getName())) {
+                    return parameter.getName();
+                }
+                return null;
+            }).filter(Objects::nonNull).collect(Collectors.toSet());
+        }
+        if (securitiesList != null) {
```

Comment:
will it ever be null? Won't it be an empty set?

Reply:
It returns only null I guess. Because the key "security" doesn't exist in the doc file. 

SECURITY LIST : null
Process finished with exit code 0

If it exist without value, the SwaggerParser throws an error : 

attribute paths.'/pet'(put).security is not of type `array`
SECURITY LIST : null

Process finished with exit code 0